scripts/deno-tests.js

```diff
@@ -1,97 +1,74 @@
-    let x = '0'
-    for (let i = 0; i < 1000; i++)x += '+' + i
-    x += ','
-    let y = 'return['
-    for (let i = 0; i < 1000; i++)y += x
-    y += ']'
-    const result = await esbuild.build({
-      stdin: {
-        contents: y,
-      },
-      write: false,
-      minify: true,
-    })
-    asserts.assertStrictEquals(result.outputFiles[0].text, y.slice(0, -2) + '];\n')
-  },
-
-  async basicTransform() {
-    const ts = 'let x: number = 1+2'
-    const result = await esbuild.transform(ts, { loader: 'ts' })
-    asserts.assertStrictEquals(result.code, 'let x = 1 + 2;\n')
-  },
+try {
+  Deno.removeSync(rootTestDir, { recursive: true })
+} catch {
+}
+Deno.mkdirSync(rootTestDir, { recursive: true })
 
-  async largeTransform() {
-    // This should be large enough to use the file system passing optimization
-    let x = '0'
-    for (let i = 0; i < 1000; i++)x += '+' + i
-    x += ','
-    let y = 'return['
-    for (let i = 0; i < 1000; i++)y += x
-    y += ']'
-    const result = await esbuild.transform(y, { minify: true })
-    asserts.assertStrictEquals(result.code, y.slice(0, -2) + '];\n')
-  },
+function test(name, fn) {
+  let testDir = path.join(rootTestDir, name)
+  Deno.test(name, async () => {
+    await Deno.mkdir(testDir, { recursive: true })
+    try {
+      await fn({ testDir })
+    } finally {
+      await Deno.remove(testDir, { recursive: true }).catch(() => null)
+      esbuild.stop()
+    }
+  })
 }
 
-async function main() {
+window.addEventListener("unload", () => {
   try {
     Deno.removeSync(rootTestDir, { recursive: true })
   } catch {
+    // root test dir possibly already removed, so ignore
   }
-  Deno.mkdirSync(rootTestDir, { recursive: true })
+}) 
 
-  // Run all tests concurrently
-  const runTest = async ([name, fn]) => {
-    let testDir = path.join(rootTestDir, name)
-    try {
-      await Deno.mkdir(testDir, { recursive: true })
-      await fn({ testDir })
-      await Deno.remove(testDir, { recursive: true }).catch(() => null)
-      return true
-    } catch (e) {
-      console.error(`❌ ${name}: ${e && e.stack || e}`)
-      return false
-    }
-  }
-  const allTestsPassed = (await Promise.all(Object.entries(tests).map(runTest))).every(success => success)
-  esbuild.stop()
+test("basicBuild", async ({ testDir }) => {
+  const input = path.join(testDir, 'in.ts')
+  const dep = path.join(testDir, 'dep.ts')
+  const output = path.join(testDir, 'out.ts')
+  await Deno.writeTextFile(input, 'import dep from "./dep.ts"; export default dep === 123')
+  await Deno.writeTextFile(dep, 'export default 123')
+  await esbuild.build({
+    entryPoints: [input],
+    bundle: true,
+    outfile: output,
+    format: 'esm',
+  })
+  const result = await import(path.toFileUrl(output))
+  asserts.assertStrictEquals(result.default, true)
+})
 
-  if (!allTestsPassed) {
-    console.error(`❌ deno tests failed`)
-    Deno.exit(1)
-  } else {
-    console.log(`✅ deno tests passed`)
```

Comment:
I had this in there because I run all tests in parallel and without this it's hard to see which top-level test failed. That's ok though; I can add this back.